desktop-widgets/diveplanner.cpp

```diff
@@ -195,7 +199,15 @@ void PlannerSettingsWidget::setDiveMode(int mode)
 
 void DivePlannerWidget::setSalinity(int salinity)
 {
-	ui.salinity->setValue(salinity / 10000.0);
+	if (salinity == int(ui.waterType->itemData(0).toDouble()*10000))
```

Comment:
For extensibility, could we do this in a loop [find matching entry, whereby the last (null) entry matches everything]?

Reply:
Ok.